doc/release-notes/7700-upgrade-payara.md

```diff
@@ -0,0 +1,6 @@
+### Payara 5.2021.1 (or Higher) Required
+
+Some changes in this release require an upgrade to Payara 5.2021.1 or higher.
+
+Instructions on how to update can be found in the
+[Payara documentation](https://docs.payara.fish/community/docs/5.2021.1/documentation/user-guides/upgrade-payara.html)
```

Comment:
In the 5.3 release notes ( https://github.com/IQSS/dataverse/releases/tag/v5.3 ) we said:

> It would likely be safer to upgrade Payara first, while still running Dataverse 5.2, and then proceed with the steps below. Upgrading from an earlier version of Payara should be a straightforward process: Undeploy Dataverse; stop Payara; move the current Payara directory out of the way; unzip the new Payara version in its place; replace the brand new payara/glassfish/domains/domain1 with your old, preserved domain1; start Payara, deploy Dataverse 5.2. We still recommend that you read the detailed upgrade instructions above; and, if you run into any issues with this upgrade, it will help to be able to separate them from any problems with the upgrade of Dataverse proper.

@poikilotherm how do you feel about putting equivalent text here? We can always remove it, especially at release time, if we decide we don't need it.

Reply:
Ok. If we merge the other PR containing changes only compatible with an upgraded Payara, we should mention updating the environment first before deploying the new WAR.

Reply:
@djbrooke mention this, but I am wary of upgrading Payara too often. I get that we took way too long to upgrade from Glassfish 4, but we don't want every release to include a Payara upgrade. (we should try to limit them to some more standard schedule - once a year, maybe 1 every 6 months). Unless of course there is a critical security fix.

Reply:
@poikilotherm sure, done in 4e94a8015

@scolapasta sure, but I saw "Unblocks #7245, #7695 and maybe #7680" and figured it's probably worth it. I'll put it in QA but please feel free to pull it out.

Reply:
Right, but one is a draft, one is not yet worked on, and the third we have a solution for that does not require an app server upgrade. So I don't really want this to get merged, at least not for 5.4 and then to decide when we want to have our next infrastructure upgrade like this.

Don't get me wrong, I want us to upgrade Payara, I just want us to manage it well, as it does require overhead for developers, for reviewers, for QA.

Let's have a tech hours discussion this week or next on it.